src/PetriEngine/Colored/ColoredPetriNetBuilder.cpp

```diff
@@ -117,8 +117,14 @@ namespace PetriEngine {
         assert(t < _transitions.size());
         assert(p < _places.size());
 
-        if(input) _places[p]._post.emplace_back(t);
-        else      _places[p]._pre.emplace_back(t);
+        if(input) {
+            _places[p]._post.emplace_back(t);
+            std::sort(_places[p]._post.begin(), _places[p]._post.end());
+        }
+        else {
+            _places[p]._pre.emplace_back(t);
+            std::sort(_places[p]._pre.begin(), _places[p]._pre.end());
+        }
```

Comment:
Jeg frygter lidt, at sorteringen her er dyr, da det sker for hvert eneste place/transition/arc. På den anden side er der kun et element der skal flyttes på plads 🤔